src/services/ipfs/backend.rs

```diff
@@ -0,0 +1,162 @@
+use super::builder::Builder;
+use super::dir_stream::DirStream;
+use super::request_writer::{RequestWriter, IpfsWriterFuture};
+
+use crate::{Accessor, AccessorMetadata, BytesReader, BytesWriter, ObjectMetadata, ObjectMode, DirStreamer};
+use crate::ops::{OpCreate, OpRead, OpWrite, OpStat, OpDelete, OpList};
+
+use std::fmt;
+use std::sync::Arc;
+use std::io;
+use async_trait::async_trait;
+use bytes::{Bytes, Buf};
+use futures::TryStreamExt;
+
+use ipfs_api::response::FilesLsResponse;
+use ipfs_api::{IpfsClient, IpfsApi};
+use ipfs_api;
+
+use futures::channel::mpsc::{self};
+use metrics::increment_counter;
+use minitrace::trace;
+
+/// Backend for IPFS service
+#[derive(Clone)]
+pub struct Backend {
+  root: String,
+  client: IpfsClient,
+}
+
+impl fmt::Debug for Backend {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    f.debug_struct("ipfs::Backend").field("root", &self.root).finish()
+  }
+}
+
+impl Backend {
+    /// Constructor for ipfs builder
+    pub fn new(root: String) -> Self {
+      Self { root, client: IpfsClient::default() }
```

Comment:
This will only allow connecting the default endpoint `localhost:5001`. Maybe we should allow users to specify the endpoint for the ipfs client?

Maybe using https://docs.rs/ipfs-api-backend-hyper/latest/ipfs_api_backend_hyper/trait.TryFromUri.html#method.from_str ?